File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.java
class Solution {
    public int minimumBoxes (int [] apple, int [] capacity) {
        int n = apple.length, m = capacity.length;
        int sumApples = 0, count = 0;
        for (int i = 0; i < n; i++)
            sumApples += apple [i];
        Arrays.sort (capacity);
        int right = m - 1;
        while (right >= 0) {
            sumApples -= capacity [right];
            count++;
            if (sumApples <= 0) break;
            right--;
        }
        return count;
    }
}

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes (self, apple : List [int], capacity : List [int]) -> int:
        total = sum (apple)
        capacity.sort (reverse = True)
        count = 0
        for c in capacity:
            total -= c
            count += 1
            if total <= 0: break
        return count